Exercises/NoModules/Chapter 13/Soln13_05/Box.h

```diff
@@ -0,0 +1,52 @@
+#ifndef BOX_H
+#define BOX_H
+
+#include <compare>  // For std::partial_ordering (see Chapter 4)
+#include <ostream>  // For std::ostream
+#include <format>
+
+class Box
+{
+public:
+  // Constructors
+  Box() = default;
+  Box(double l, double w, double h) : m_length{ l }, m_width{ w }, m_height{ h } {}
+
+  double volume() const { return m_length * m_width * m_height; }
+
+  // Accessors
+  double getLength() const { return m_length; }
+  double getWidth()  const { return m_width; }
+  double getHeight() const { return m_height; }
+
+  std::partial_ordering operator<=>(const Box& otherBox) const 
+  { 
+    return volume() <=> otherBox.volume();
+  }
+  std::partial_ordering operator<=>(double otherVolume) const 
+  { 
+    return volume() <=> otherVolume;
+  }
+
+  bool operator==(const Box& otherBox) const = default;
+
+  // Unary negation operator (!box is true if the Box has no volume)
+  bool operator!() const { return volume() == 0; }  
```

Comment:
This operator is not needed.
Just implementing the bool() conversion op is enough for !box to work.

Reply:
ok, that seemed to be the goal of this exercises, after reading the next exercise...

Reply:
Yeah, didn't want to everything at once. At least this way they also have one more exercise on implementing a unary operator.